tests/python_package_test/test_engine.py

```diff
@@ -2278,6 +2297,98 @@ def test_multiple_feval_train():
     assert 'decreasing_metric' in evals_result['valid_0']
 
 
+def test_objective_callable_train_binary_classification():
+    X, y = load_breast_cancer(return_X_y=True)
+    params = {
+        'verbose': -1,
+        'objective': logloss_obj,
+        'learning_rate': 0.01
+    }
+    train_dataset = lgb.Dataset(X, y)
+    booster = lgb.train(
+        params=params,
+        train_set=train_dataset,
+        num_boost_round=20
+    )
+    y_pred = logistic_sigmoid(booster.predict(X))
+    logloss_error = log_loss(y, y_pred)
+    rocauc_error = roc_auc_score(y, y_pred)
+    assert booster.params['objective'] == 'none'
+    assert logloss_error == pytest.approx(0.55, 0.1)
+    assert rocauc_error == pytest.approx(0.99, 0.5)
+
+
+def test_objective_callable_train_regression():
+    X, y = make_synthetic_regression()
+    params = {
+        'verbose': -1,
+        'objective': mse_obj
+    }
+    lgb_train = lgb.Dataset(X, y)
+    booster = lgb.train(
+        params,
+        lgb_train,
+        num_boost_round=20
+    )
+    y_pred = booster.predict(X)
+    mse_error = mean_squared_error(y, y_pred)
+    assert booster.params['objective'] == 'none'
+    assert mse_error == pytest.approx(286, 1)
+
+
+def test_objective_callable_cv_binary_classification():
+    X, y = load_breast_cancer(return_X_y=True)
+    params = {
+        'verbose': -1,
+        'objective': logloss_obj,
+        'learning_rate': 0.01
+    }
+    train_dataset = lgb.Dataset(X, y)
+    cv_res = lgb.cv(
+        params,
+        train_dataset,
+        num_boost_round=20,
+        nfold=3,
+        return_cvbooster=True
+    )
+    cv_booster = cv_res['cvbooster'].boosters
+    cv_logloss_errors = [
+        log_loss(y, logistic_sigmoid(cb.predict(X))) < 0.56 for cb in cv_booster
+    ]
+    cv_objs = [
+        cb.params['objective'] == 'none' for cb in cv_booster
+    ]
+    assert all(cv_objs)
+    assert all(cv_logloss_errors)
+
+
+def test_objective_callable_cv_regression():
+    X, y = make_synthetic_regression()
+    lgb_train = lgb.Dataset(X, y)
+    params_with_metric = {
+        'verbose': -1,
+        'objective': mse_obj,
+        'metric': 'l2'
+    }
+    cv_res = lgb.cv(
+        params_with_metric,
```

Comment:
Looks like `metric` param isn't used in this test.
```suggestion
    params = {
        'verbose': -1,
        'objective': mse_obj
    }
    cv_res = lgb.cv(
        params,
```

Reply:
got it